src/data-layer/db-connector.ts

```diff
@@ -1,19 +1,87 @@
-import products from '../../mocks/products'
 import { Product } from "../types/api-types";
+import { Client } from "pg";
+import { config } from "../utils/getDBClientConfig";
+
 export const DbConnector = {
     getById: (productId: Product["id"]): Promise<Product | null> => {
-        return new Promise((resolve, reject) => {
-            const product = products.find(({ id }) => productId === id);
-            if (product) {
-                resolve(product);
-            } else {
-                reject(null);
+        const client = new Client({
+            ...config,
+        });
+        return new Promise(async (resolve, reject) => {
+            try {
+                await client.connect();
+                console.log(`Fetching product with id ${productId}`);
+                const getProductQuery =
+                    "SELECT * FROM products INNER JOIN stocks ON products.id = stocks.product_id WHERE products.id=$1";
+                const {rows: products} = await client.query(getProductQuery, [productId]);
+                const product:Product = products.pop();
+                resolve(product ? product : null);
+            } catch (e) {
+                const errorMessage = e.message || e;
+                console.error("ERROR: Can't fetch the product:", errorMessage);
+                reject(errorMessage);
+                await client.query("ROLLBACK");
+            } finally {
+                await client.end();
+                console.log(`Connection to the DB was closed`);
             }
         })
     },
     getAll: (): Promise<Product[]> => {
-        return new Promise((resolve, reject) => {
-            resolve(products);
+        const client = new Client({
+            ...config,
+        });
+        return new Promise(async (resolve, reject) => {
+            try {
+                await client.connect();
+                console.log(`Getting all products`);
+                const getProductsQuery =
+                    "SELECT * FROM products INNER JOIN stocks ON products.id = stocks.product_id";
+                const { rows: products } = await client.query(getProductsQuery);
+
+                console.log("Products were successfully fetched");
+                resolve(products);
+            } catch (e) {
+                const errorMessage = e.message || e;
+                console.error("ERROR: can't fetch all products:", errorMessage);
+                reject(errorMessage);
+                await client.query("ROLLBACK");
+            } finally {
+                await client.end();
+                console.log(`Connection to the DB was closed`);
+            }
         })
-    }
+    },
+    createProduct: (product: Product): Promise<Product['id']> => {
+        const client = new Client({
```

Comment:
Does it need to create the client in each method separately?
Is it possible to create it once before export const DbConnector?

Reply:
That was my first attempt, and then I faced errors. So one of the solutions was to create a connection per each request or use a connection pool. As you can see, I have had to choose the first one, to keep focus on general task.